File: C#/Tuitorial-0/neuralNetwork0.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        // Input vector
        List<double> inputs = new List<double> { 1.0, 2.0, 3.0, 2.5 };

        // Initialize weights randomly using the Random class
        Random rand = new Random();
        List<double> weights1 = new List<double> { rand.NextDouble(), rand.NextDouble(), rand.NextDouble(), rand.NextDouble() };
        List<double> weights2 = new List<double> { rand.NextDouble(), rand.NextDouble(), rand.NextDouble(), rand.NextDouble() };
        List<double> weights3 = new List<double> { rand.NextDouble(), rand.NextDouble(), rand.NextDouble(), rand.NextDouble() };

        // Initialize biases as 0
        double bias1 = 0;
        double bias2 = 0;
        double bias3 = 0;

        // Calculate the output
        List<double> output = new List<double>
        {
            (inputs[0] * weights1[0]
             + inputs[0] * weights1[1]
             + inputs[0] * weights1[2]
             + inputs[0] * weights1[3]
             + bias1),
            (inputs[0] * weights2[0]
             + inputs[0] * weights2[1]
             + inputs[0] * weights2[2]
             + inputs[0] * weights2[3]
             + bias2),
            (inputs[0] * weights3[0]
             + inputs[0] * weights3[1]
             + inputs[0] * weights3[2]
             + inputs[0] * weights3[3]
             + bias3)
        };

        // Output the result
        Console.WriteLine("The output is: [{0}, {1}, {2}]",
            Math.Round(output[0], 4),
            Math.Round(output[1], 4),
            Math.Round(output[2], 4));
    }
}

File: Python/Tuitorial-0/neuralnetwork0.py
from typing import List
import numpy as np

"""
This script demonstrates how a neuron works by performing calculations with inputs, 
weights, and biases. It creates a simple neural network with 3 neurons.
"""

# Inputs, weights, and biases for a neural network with 3 neurons.
inputs: List[float] = [1.0, 2.0, 3.0, 2.5]  # initializing inputs

# Initializing weights randomly for each neuron
weights1: List[float] = np.random.rand(4)
weights2: List[float] = np.random.rand(4)
weights3: List[float] = np.random.rand(4)

# Biases initialization as 0
bias1: float = 0
bias2: float = 0
bias3: float = 0

# Calculate the output for each neuron
output: List[float] = [
    (inputs[0] * weights1[0] + inputs[1] * weights1[1] + inputs[2] * weights1[2] +
     inputs[3] * weights1[3] + bias1),
    (inputs[0] * weights2[0] + inputs[1] * weights2[1] + inputs[2] * weights2[2] +
     inputs[3] * weights2[3] + bias2),
    (inputs[0] * weights3[0] + inputs[1] * weights3[1] + inputs[2] * weights3[2] +
     inputs[3] * weights3[3] + bias3),
]

# Print rounded output values
print(f"The output is: {np.round(output, 4)}")
